internal/consensus/state.go

```diff
@@ -2440,11 +2442,18 @@ func proposerWaitTime(lt tmtime.Source, h types.Header) time.Duration {
 // The result of proposalStepWaitingTime is compared with the configured `timeout-propose` duration,
 // and the validator waits for whichever duration is larger before advancing to the next step
 // and prevoting nil.
-func proposalStepWaitingTime(lt tmtime.Source, h types.Header, tp types.TimestampParams) time.Duration {
+func proposalStepWaitingTime(lt tmtime.Source, bt time.Time, tp types.TimestampParams) time.Duration {
 	t := lt.Now()
-	wt := h.Time.Add(tp.Precision).Add(tp.MsgDelay)
+	wt := bt.Add(tp.Precision).Add(tp.MsgDelay)
 	if t.After(wt) {
 		return 0
 	}
 	return wt.Sub(t)
 }
+
+func maxDuration(d1 time.Duration, d2 time.Duration) time.Duration {
+	if d1-d2 > 0 {
+		return d1
+	}
+	return d2
```

Comment:
Is there some reason not to compare these directly?
```suggestion
	if d1 >= d2 {
		return d1
	}
	return d2
```